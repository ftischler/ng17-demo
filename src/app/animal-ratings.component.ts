import { ChangeDetectionStrategy, Component } from '@angular/core';
import { KeyValuePipe } from '@angular/common';

@Component({
  selector: 'ng17-demo-animal-ratings',
  standalone: true,
  imports: [KeyValuePipe],
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <ul>
      @for (animalRating of animalRatings | keyvalue; track animalRating.key) {
      <li>{{ animalRating.key }}: {{ animalRating.value }}</li>
      }
    </ul>
  `,
})
export class AnimalRatingsComponent {
  readonly animalRatings = {
    bear: 10,
    dog: 10,
    cat: 10,
  };
}
