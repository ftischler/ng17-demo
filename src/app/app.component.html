<ng17-demo-nx-welcome></ng17-demo-nx-welcome> <router-outlet></router-outlet>
