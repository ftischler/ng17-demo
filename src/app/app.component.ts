import { ChangeDetectionStrategy, Component, signal } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AnimalRatingsComponent } from './animal-ratings.component';

@Component({
  standalone: true,
  imports: [AnimalRatingsComponent, RouterModule],
  selector: 'ng17-demo-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent {
  loggedIn = signal<boolean>(false);
  longText = 'This is not a long text';

  login() {
    this.loggedIn.set(true);
  }

  logout() {
    this.loggedIn.set(false);
  }
}
