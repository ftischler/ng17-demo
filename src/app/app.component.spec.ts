import { DeferBlockState, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent, RouterTestingModule],
    }).compileComponents();
  });

  it('should render title', async () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const button: HTMLButtonElement = fixture.debugElement.query(
      By.css('button')
    ).nativeElement;
    button.click();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();

    const [animalRatingsFixture] = await fixture.getDeferBlocks();
    await animalRatingsFixture.render(DeferBlockState.Complete);

    expect(fixture).toMatchSnapshot();
  });
});
