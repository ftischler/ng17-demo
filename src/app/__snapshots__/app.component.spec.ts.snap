// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render title 1`] = `
<ng17-demo-root
  loggedIn={[Function Function]}
  longText={[Function String]}
>
  <button
    type="button"
  >
    Logout
  </button><div>
    <p>
      This is not a long text
    </p>
    <span>
      Placeholder
    </span>
  </div>
</ng17-demo-root>
`;

exports[`AppComponent should render title 2`] = `
<ng17-demo-root
  loggedIn={[Function Function]}
  longText={[Function String]}
>
  <button
    type="button"
  >
    Logout
  </button><div>
    <p>
      This is not a long text
    </p>
    <ng17-demo-animal-ratings>
      <ul>
        <li>
          <div>
            üêª Bear
          </div>
          <div>
            10
          </div>
        </li>
        <li>
          <div>
            üê∂ Dog
          </div>
          <div>
            10
          </div>
        </li>
        <li>
          <div>
            üê± Cat
          </div>
          <div>
            10
          </div>
        </li>
        <li>
          <div>
            üï∑ Spider
          </div>
          <div>
            2
          </div>
        </li>
      </ul>
    </ng17-demo-animal-ratings>
  </div>
</ng17-demo-root>
`;
